wallet/wallet_test.go

```diff
@@ -4,26 +4,62 @@ import (
 	"encoding/hex"
```

Comment:
Added tests for signature verification, plus some refactoring to make things clearer.